homeassistant/helpers/event.py

```diff
@@ -86,15 +88,19 @@ def point_in_time_listener(event):
         # listener gets lined up twice to be executed. This will make
         # sure the second time it does nothing.
         point_in_time_listener.run = True
+        async_remove()
+
+        hass.async_add_job(action, now)
 
-        def fire_action():
-            """Run the point in time listener action."""
-            remove()
-            action(now)
+    future = run_callback_threadsafe(
+        hass.loop, hass.bus.async_listen, EVENT_TIME_CHANGED,
+        point_in_time_listener)
+    async_remove = future.result()
```

Comment:
This should be wrapped in a lambda, calling result() now means it blocks for it right now.